plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -741,8 +741,8 @@ else if (orcColumn != null) {
                     }
                 }
             }
-            if (e instanceof TrinoException) {
-                throw (TrinoException) e;
+            if (e instanceof TrinoException ex) {
+                throw ex;
```

Comment:
How did you decide the scope of this change? For instance, `TypeConverter` has the below code. Also, my IntelliJ doesn't blame this code as redundant cast. 
```java
        if (type instanceof DecimalType) {
            return fromDecimal((DecimalType) type);
        }
```

Reply:
Yes right, My IntelliJ also does not blame the redundant cast. @findinpath has pointed out the code.

I assume the warning comes from the code where we initialize a variable with cast and use it.

```java
        if (type instanceof DecimalType) {
            DecimalType decimalType = (DecimalType) type;
            return fromDecimal(decimalType);
        }
```

@findinpath is there a way to find this kind of code snippet in the codebase?

Reply:
No actual magic behind my finding. I just looked for through all `instanceof` occurrences within `trino-iceberg` if there is any other place where pattern matching may come in handy.

Reply:
Thanks, @findinpath, I try to find other places as well in iceberg connector and modified the code.

Reply:
@findinpath @krvikash Let's avoid wasting time on this minor refactoring. Since it's little hard to find the usage, there's no need to change in my opinion.

Reply:
Sure @ebyhr, I'll keep the change only for those where IntelliJ gives a warning message.

Reply:
Kept the changes that give warning in IntelliJ. Other changes have been reverted.